treetraversal.py

```
@@ -0,0 +1,129 @@
+
+    def render(self):
+        pass
+
+class WidgetNode:
+    def __init__(self, props, children=None):
+        self.props = props
+        self.children = children or []
+
+Renderable = Union[BaseComponent, WidgetNode]
+
+class Renderable:
```

Ruff (0.8.2)

Redefinition of unused `Renderable` from line 39

(F811)